src/python/pants/engine/internals/parser_test.py

```diff
@@ -17,18 +18,70 @@ def test_imports_banned() -> None:
     assert "Import used in dir/BUILD at line 4" in str(exc.value)
 
 
-def test_unrecognized_symbol() -> None:
-    parser = Parser(
-        target_type_aliases=["tgt"],
-        object_aliases=BuildFileAliases(
-            objects={"obj": 0},
-            context_aware_object_factories={"caof": lambda parse_context: lambda _: None},
-        ),
+def test_unrecogonized_symbol() -> None:
+    def perform_test(tta: list, good_err: str) -> None:
```

Comment:
A more precise type hint is `list[str]`, which makes clear what the element type is.

To use this, you'll need to add `from __future__ import annotations` to the top of the file, in between the copyright and your first import. Normally, `list[str]` is invalid syntax, but this enables a syntax feature that it's legal if used in a type hint.

Reply:
Good to know there's a way to make that more precise!